miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -0,0 +1,490 @@
+        
+        // Stock columns
+        HotUtils.makeColumnForEnum('STR Status', show['Stock'], true, 'strStatus', Constants.strStatuses),
+        {
+          header: 'DNAse',
+          data: 'dnaseTreated',
+          type: 'dropdown',
+          trimDropdown: false,
+          validator: HotUtils.validator.requiredAutocomplete,
+          source: [ 'True', 'False' ],
+          unpack: function (sam, flat, setCellMeta) {
+            flat.dnaseTreated = (sam.dnaseTreated ? 'True' : 'False');
+          },
+          pack: function (sam, flat, errorHandler) {
+            sam.dnaseTreated = flat.dnaseTreated === 'True';
+          },
+          include: show['Stock'] && config.targetSampleClass.dnaseTreatable
+        },
+        HotUtils.makeColumnForFloat('Vol. (&#181;l)', (show['Stock'] || show['Aliquot']), 'volume'),
+        HotUtils.makeColumnForFloat('Conc. (ng/&#181;l)', (show['Stock'] || show['Aliquot']), 'concentration'),
+        HotUtils.makeColumnForFloat('New RIN', config.rnaSamples, 'rin'),
+        HotUtils.makeColumnForFloat('New DV200', config.rnaSamples, 'dv200'),
+        
+        // QC status columns for detailed and non-detailed samples
+        {
+          header: 'QC Passed?',
+          data: 'qcPassed',
+          type: 'dropdown',
+          trimDropdown: false,
+          source: ['Unknown', 'True', 'False'],
+          unpack: function (sam, flat, setCellMeta) {
+            if (sam.qcPassed === true) flat.qcPassed = 'True';
+            else if (sam.qcPassed === false) flat.qcPassed = 'False';
+            else flat.qcPassed = 'Unknown';
+          },
+          pack: function (sam, flat, errorHandler) {
+            if (flat.qcPassed === 'True') sam.qcPassed = true;
+            else if (flat.qcPassed === 'False') sam.qcPassed = false;
+            else sam.qcPassed = null;
+          },
+          include: !Constants.isDetailedSample
+        },
+        {
+          header: 'QC Status',
+          data: 'detailedQcStatusDescription',
+          type: 'dropdown',
+          trimDropdown: false,
+          source: (function () {
+            var statuses = Constants.detailedQcStatuses.sort(function (a, b) {
+              function statusToInt (status) {
+                if (status === true) return 0;
+                if (status === false) return 1;
+                return 2;
+              }
+              return statusToInt(a.status) - statusToInt(b.status);
+            }).map(function (s) { return s.description; });
+            statuses.unshift('Not Ready'); // can't chain this because unshift returns the length of the array
+            return statuses;
+          })(),
+          validator: HotUtils.validator.requiredText,
+          unpack: function (sam, flat, setCellMeta) {
+            flat.detailedQcStatusDescription = Utils.array.maybeGetProperty(
+                Utils.array.findFirstOrNull(
+                    Utils.array.idPredicate(sam.detailedQcStatusId), Constants.detailedQcStatuses),
```

Comment:
This came from the lab -- Danielle preferred that it show up blank so she can double-click to fill down, rather than have to change the top one and drag down to the bottom of the table fill in the others. (Handsontable double-click-to-fill only fills in empty cells)